torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -419,6 +419,56 @@ class ShapePropagator {
     setUnshapedType(cat_node);
   }
 
+  at::ScalarType scalarTypeFromJitType(const c10::TypePtr& type) {
```

Comment:
I'm going to move this to jit_type since it's used in a different file as well...